src/modules/hip/kernel/lens_correction.hpp

```diff
@@ -0,0 +1,120 @@
+    float rCoeff[6] = { distortionCoeffs.f1[0], distortionCoeffs.f1[1], distortionCoeffs.f1[4], distortionCoeffs.f1[5], distortionCoeffs.f1[6], distortionCoeffs.f1[7] };
+    float tCoeff[2] = { distortionCoeffs.f1[2], distortionCoeffs.f1[3] };
+
+    // Get the focal length and principal point of the camera
+    float u0 = cameraMatrix.f1[2],  v0 = cameraMatrix.f1[5];
+    float fx = cameraMatrix.f1[0],  fy = cameraMatrix.f1[4];
+
+    float *rowRemapTableTemp = rowRemapTable + id_z * remapTableStridesNH.x + id_y * remapTableStridesNH.y + id_x;
+    float *colRemapTableTemp = colRemapTable + id_z * remapTableStridesNH.x + id_y * remapTableStridesNH.y + id_x;
+    float xCamera = id_y * inverseMatrix.f1[1] + inverseMatrix.f1[2] + id_x * inverseMatrix.f1[0];
+    float yCamera = id_y * inverseMatrix.f1[4] + inverseMatrix.f1[5] + id_x * inverseMatrix.f1[3];
+    float zCamera = id_y * inverseMatrix.f1[7] + inverseMatrix.f1[8] + id_x * inverseMatrix.f1[6];
+    float z = 1./zCamera, x = xCamera * z, y = yCamera * z;
+    float xSquare = x * x, ySquare = y * y;
+    float r2 = xSquare + ySquare, xyMul2 = 2 * x * y;
+    float kr = (1 + ((rCoeff[2] * r2 + rCoeff[1]) * r2 + rCoeff[0]) * r2) / (1 + ((rCoeff[5] * r2 + rCoeff[4]) * r2 + rCoeff[3]) *r2);
+    float colLoc = fx * (x * kr + tCoeff[0] * xyMul2 + tCoeff[1] * (r2 + 2 * xSquare)) + u0;
+    float rowLoc = fy * (y * kr + tCoeff[0] * (r2 + 2 * ySquare) + tCoeff[1] * xyMul2) + v0;
+    *colRemapTableTemp = colLoc;
+    *rowRemapTableTemp = rowLoc;
+}
+
+// -------------------- Set 3 - Kernel Executors --------------------
+
+RppStatus hip_exec_lens_correction_tensor(RpptDescPtr dstDescPtr,
+                                          Rpp32f *rowRemapTable,
+                                          Rpp32f *colRemapTable,
+                                          RpptDescPtr remapTableDescPtr,
+                                          Rpp32f *cameraMatrix,
+                                          Rpp32f *distanceCoeffs,
+                                          Rpp32f *newCameraMatrix,
+                                          RpptROIPtr roiTensorPtrSrc,
+                                          RpptRoiType roiType,
+                                          rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = dstDescPtr->w;
+    int globalThreads_y = dstDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
+
+    float *inverseMatrix = handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem;
+    hipLaunchKernelGGL(get_inverse_hip,
+                       dim3(1, 1, ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                       dim3(1, 1, LOCAL_THREADS_Z),
+                       0,
+                       handle.GetStream(),
+                       (d_float9 *)cameraMatrix,
+                       (d_float9 *)inverseMatrix);
+    hipDeviceSynchronize();
```

Comment:
Please remove the hipDeviceSynchronize() calls here
Since launching in same stream, this calls might not be needed